File: splitpal/lib/features/receipts/domain/entities/receipt.dart
import 'package:equatable/equatable.dart';

class Receipt extends Equatable {
  final String id;
  final String imageUrl;
  final String? note;
  final DateTime receiptDate;
  final List<ReceiptTag> tags;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const Receipt({
    required this.id,
    required this.imageUrl,
    this.note,
    required this.receiptDate,
    this.tags = const [],
    required this.createdAt,
    this.updatedAt,
  });

  @override
  List<Object?> get props => [id, imageUrl, note, receiptDate, tags, createdAt, updatedAt];
}

class ReceiptTag extends Equatable {
  final String id;
  final String name;
  final String color;

  const ReceiptTag({
    required this.id,
    required this.name,
    required this.color,
  });

  @override
  List<Object?> get props => [id, name, color];
}

class ReceiptDaySummary extends Equatable {
  final String date; // YYYY-MM-DD (UTC)
  final int count;
  final List<String> thumbUrls;

  const ReceiptDaySummary({
    required this.date,
    required this.count,
    this.thumbUrls = const [],
  });

  @override
  List<Object?> get props => [date, count, thumbUrls];
}

File: server/src/type/receipt.ts
import { Types } from 'mongoose';

export interface ReceiptInput {
    imageUrl: string;
    totalAmount?: number;
    note?: string;
    tagIds: string[];
    receiptDate?: string; // ISO date string YYYY-MM-DD
}

export interface ReceiptUpdateInput {
    totalAmount?: number;
    note?: string;
    tagIds?: string[];
}

export interface ReceiptDto {
    id: string;
    imageUrl: string;
    totalAmount: number;
    note?: string | null;
    tags: TagDto[];
    receiptDate: string; // ISO date (UTC)
    createdAt: string;
    updatedAt: string;
}

export interface TagDto {
    id: string;
    name: string;
    color: string;
}

export interface MonthSummaryItem {
    date: string; // YYYY-MM-DD
    count: number;
    totalAmount: number;
    thumbUrls: string[];
}

export interface DayReceiptsResponse {
    date: string;
    receipts: ReceiptDto[];
}

export type ObjectIdLike = string | Types.ObjectId;
